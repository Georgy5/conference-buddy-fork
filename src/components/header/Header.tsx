import React from "react"
import { StaticImage } from "gatsby-plugin-image"

function Header() {
  return (
    <header className="mb-3">
      <nav className="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between">
        <div className="d-none d-sm-block d-flex align-items-center">
          <a href="/" title="Startpage">
            <StaticImage
              src="../../images/LogoConferenceBuddy_head_small.png"
              alt="Logo"
            />
          </a>
        </div>
      </nav>
    </header>
  )
}
export default Header
