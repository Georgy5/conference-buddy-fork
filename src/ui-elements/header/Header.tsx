import React, { ReactElement } from "react"
import { StaticImage } from "gatsby-plugin-image"
import { Link } from "gatsby"
import { SignIn } from "../../domain/sign-in/SignIn"

function Header(): ReactElement {
  return (
    <header className="mb-3">
      <nav className="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between">
        <div className="d-flex align-items-center">
          <Link to="/" title="Startpage">
            <StaticImage
              src="../../assets/images/LogoConferenceBuddy.png"
              alt="Conference Buddy logo"
              height={50}
              loading="eager"
              placeholder="none"
            />
          </Link>
          <div className="text-white d-block d-sm-none">Conference Buddy</div>
          <div className="text-white d-none d-sm-block fs-3">
            Conference Buddy
          </div>
        </div>
        <SignIn />
      </nav>
    </header>
  )
}
export { Header }
