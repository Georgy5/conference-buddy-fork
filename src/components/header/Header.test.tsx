import { render, screen, cleanup, RenderResult } from "@testing-library/react"
import Header from "./Header"

describe("Header", () => {
  let component: RenderResult
  beforeAll(() => {
    component = render(<Header />)
  })

  afterAll(cleanup)

  it("renders the header element", () => {
    const header = screen.getByRole("banner")
    expect(header).toBeVisible()
  })

  it("renders all necessary elements", () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
