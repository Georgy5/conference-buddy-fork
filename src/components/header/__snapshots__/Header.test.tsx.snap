// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders all necessary elements 1`] = `
<DocumentFragment>
  <header
    class="mb-3"
  >
    <nav
      class="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between"
    >
      <div
        class="d-none d-sm-block d-flex align-items-center"
      >
        <a
          href="/"
          title="Startpage"
        >
          <img
            alt="Logo"
            src="../../images/LogoConferenceBuddy_head_small.png"
          />
        </a>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
